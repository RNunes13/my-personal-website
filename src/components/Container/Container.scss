
@import '../../assets/scss/settings/_config-base';

.rn-container {
  @include rem(padding-left, 20px);
  @include rem(padding-right, 20px);
  width: 100%;
  margin: 0 auto;
  transition-property: padding, width;
  transition-duration: $transition-duration;
  transition-timing-function: $transition-function;

  @include breakpoint(min-width, 1200px) {
    @include rem(max-width, 1140px);
  }

  @include between-breakpoints(992px, 1199px) {
    @include rem(max-width, 960px);
  }

  @include between-breakpoints(768px, 991px) {
    @include rem(max-width, 720px);
  }

  @include between-breakpoints(576px, 767px) {
    @include rem(max-width, 540px);
  }
  
  @include breakpoint(max-width, 575px) {
    @include rem(padding-left, 15px);
    @include rem(padding-right, 15px);
    max-width: none;
  }
}
