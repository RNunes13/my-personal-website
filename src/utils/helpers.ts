/**
 * Get a RGB color from the hexadecimal.
 * @param {string} hex Hex value in the format `#181923`.
 * @public
 */
export const getRgbFromHex = (hex = '') => {
  const hexCode = hex.replace('#', '')

  if (hexCode.length !== 6) {
    throw Error('Only six-digit hex colors are allowed.')
  }

  const rgbHex = hexCode.match(/.{1,2}/g) as RegExpMatchArray

  return [
    parseInt(rgbHex[0], 16),
    parseInt(rgbHex[1], 16),
    parseInt(rgbHex[2], 16),
  ]
}
