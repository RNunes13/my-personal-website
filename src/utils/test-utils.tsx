import { FC, ReactElement } from 'react'
import { ThemeProvider } from '@emotion/react'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import renderer, { ReactTestRenderer } from 'react-test-renderer'
import userEvent from '@testing-library/user-event'

import { theme } from 'styles/theme/Theme'

interface IProviders {
  children: ReactElement
}

const Providers: FC<IProviders> = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

export const renderWithTheme = (component: ReactElement): ReactTestRenderer =>
  renderer.create(<Providers>{component}</Providers>)

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
): RenderResult => render(ui, { wrapper: Providers, ...options })

const renderWithUser = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
) => ({
  user: userEvent.setup(),
  ...render(ui, { wrapper: Providers, ...options }),
})

export * from '@testing-library/react'
export * from '@testing-library/jest-dom'
export { customRender as render }
export { renderWithUser }
