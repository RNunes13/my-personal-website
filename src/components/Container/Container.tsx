
import * as React from 'react';
import classNames from 'classnames';

import './Container.scss';

interface ContainerProps {
  className?: string;
  style?: React.CSSProperties | undefined;
};

const Container: React.FunctionComponent<ContainerProps> = ({ className, style, children }) => {
  return (
    <div className={ classNames('rn-container', className) } style={ style }>
      { children }
    </div>
  )
}

export default Container;
