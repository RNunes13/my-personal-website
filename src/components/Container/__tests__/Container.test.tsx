import { renderWithTheme } from 'utils/test-utils'
import { Container } from '../Container'
import type { ContainerProps } from '../Container'

const defaultProps: ContainerProps = {
  children: <div data-testid="container-children">Container children</div>,
}

const getComponent = (props: Partial<ContainerProps> = {}) => (
  <Container {...defaultProps} {...props} />
)

describe('Container unit test', () => {
  it('should render container with default grid (1290)', () => {
    const tree = renderWithTheme(getComponent()).toJSON()
    expect(tree).toMatchSnapshot('Container - Grid default (1290)')
  })

  it('should render container with grid of 940', () => {
    const tree = renderWithTheme(getComponent({ grid: 'grid940' })).toJSON()
    expect(tree).toMatchSnapshot('Container - Grid 940')
  })
})
