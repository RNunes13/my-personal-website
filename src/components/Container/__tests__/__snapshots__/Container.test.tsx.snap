// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button unit test should render container with default grid (1290): Container - Grid default (1290) 1`] = `
.emotion-0 {
  width: 100%;
  box-sizing: border-box;
  position: relative;
  padding-right: 30px;
  padding-left: 30px;
  margin-right: auto;
  margin-left: auto;
  max-width: 1350px;
}

@media only screen and (max-width: 768px) {
  .emotion-0 {
    padding-right: 20px;
    padding-left: 20px;
  }
}

<div
  className="emotion-0 emotion-1"
>
  <div
    data-testid="container-children"
  >
    Container children
  </div>
</div>
`;

exports[`Button unit test should render container with grid of 940: Container - Grid 940 1`] = `
.emotion-0 {
  width: 100%;
  box-sizing: border-box;
  position: relative;
  padding-right: 30px;
  padding-left: 30px;
  margin-right: auto;
  margin-left: auto;
  max-width: 1000px;
}

@media only screen and (max-width: 768px) {
  .emotion-0 {
    padding-right: 20px;
    padding-left: 20px;
  }
}

<div
  className="emotion-0 emotion-1"
>
  <div
    data-testid="container-children"
  >
    Container children
  </div>
</div>
`;
